### ISSUE TITLE
Subsequent variable declarations must have the same type.  Variable 'module' must be of type '{ id: string; }', but here has type 'NodeModule

### ISSUE THREAD

**Issue by crs117** [cj-wang/mean-start#7]
First of all cj-wang thanks for this repo.  

It seems as though there is constant churn with typescript and ang-cli. The current repo includes the file https://github.com/cj-wang/mean-start/blob/master/client/typings.d.ts which contains

`/* SystemJS module definition */
declare var module: {
  id: string;
};`

When building i would get "/node_modules/@types/node/index.d.ts (85,13): Subsequent variable declarations must have the same type.  Variable 'module' must be of type '{ id: string; }', but here has type 'NodeModule". So module was declared twice but one requires an id: string the node type does not. 

In order to build I had to comment out the https://github.com/cj-wang/mean-start/blob/master/client/typings.d.ts file entirely to get it to build. Is this file needed or did i mess up something else? I am new to typescript and ang-cli.

**Comment by cj-wang** [comment]
`ng build` and `npm run build` both work fine as I tested.
Do you use `tsc` directly for building? If so, make sure `node_modules` is excluded. 
Please let me know.

Anyway this file had been problematic and was just fixed:
https://github.com/angular/angular-cli/commit/708719585e09571053fcdbe7a37428fe98806c8d

**Comment by crs117** [comment]
I was following your instructions so I was using `npm run build`. I am very new to angular and typescript so please be patient with me.

[angular/angular-cli@7087195](https://github.com/angular/angular-cli/commit/708719585e09571053fcdbe7a37428fe98806c8d) seems to fix the issue. I am unsure why this did not update on my machine because i had updated to @angular/cli": "1.0.0-rc.4 and this looks to be an included fix. 

Either way I reverted everything to your original package which runs @angular/cli": "1.0.0-rc.2, re-ran npm install and then manually updated the client/typings.d.ts file to match the merge from the 708719 issue and everything seems to work. 

Which rc are you currently running which allowed this to build?

**Comment by cj-wang** [comment]
Strange. I'm with rc-2.
The current file on your machine was from my repo, it couldn't be updated by upgrading ng cli.
If you use `ng new PROJECT_NAME` to make a new project, you'll find the updated file.

**Comment by crs117** [comment]
I made a mistake. I just ran `ng -v` and discovered I was still running rc4. This means that you will probably need to adapt the change from [angular/angular-cli@7087195](https://github.com/angular/angular-cli/commit/708719585e09571053fcdbe7a37428fe98806c8d) when the final release comes out.

I downgraded to rc2, reverted to the original file and it builds fine with the original file.

Thanks for your help. Do I close this out?
